IPython/utils/path.py

```diff
@@ -437,6 +437,13 @@ def ensure_dir_exists(path, mode=0o755):
 
     The default permissions are 755, which differ from os.makedirs default of 777.
     """
+    path_exists = False
+    try:
+        path_exists = os.path.exists(path)
+    except UnicodeEncodeError:
+        path = path.encode("utf8")
```

Comment:
The encoding should be `sys.getfilesystemencoding() or 'utf-8'`, not unconditionally utf8. This second call can also raise, in which case False should probably be returned.